File: 프로그래머스/연습문제/Lv1/기사단원의 무기/기사단원의 무기.cpp
#include <string>
#include <vector>
#include <algorithm>
using namespace std;

int solution(int number, int limit, int power) {
    vector<int> div(number+1, 2);
    div[1] = 1;
    
    for(int i=2; i<=number; i++)
        for(int j=i*2; j<=number; j+=i)
            div[j]++;
    
    int answer = 0;
    for(int i=1; i<=number; i++){
        if(div[i] > limit) answer += power;
        else answer += div[i];
    }
    
    return answer;
}

File: 프로그래머스/연습문제/Lv1/기사단원의 무기/기사단원의 무기.py
def solution(number, limit, power):
    div = [2 for i in range(number+1)]
    div[1] = 1

    for i in range(2, number+1):
        for j in range(i*2, number+1, i):
            div[j] += 1

    answer = 0
    for i in range(1, number+1):
        if div[i] > limit: answer += power
        else: answer += div[i]

    return answer
